File: codeforces/low_difficulty/800_points/61.team_olympiad_[490A]/team_olympiad.py
# Author: José Rodolfo (jric2002)
n = int(input())
student_skills = list(map(int, input().split(" ")))
programming_students = []
maths_students = []
pe_students = []
number_student_skill = []
for i in range(0, n):
  t = student_skills[i]
  if (t == 1):
    programming_students.append(i + 1)
  elif (t == 2):
    maths_students.append(i + 1)
  else:
    pe_students.append(i + 1)
number_student_skill.append(len(programming_students))
number_student_skill.append(len(maths_students))
number_student_skill.append(len(pe_students))
number_student_skill.sort()
w = number_student_skill[0]
print(w)
for i in range(0, w):
  print(programming_students[i], maths_students[i], pe_students[i], sep=" ")

File: codeforces/low_difficulty/difficulty_800/61.team_olympiad_[490A]/team_olympiad.cpp
#include <iostream>
#include <vector>
#include <algorithm>
/* Author: José Rodolfo (jric2002) */
using namespace std;
int main() {
  unsigned short int n, t, w;
  vector<unsigned short int> programming_students;
  vector<unsigned short int> maths_students;
  vector<unsigned short int> pe_students;
  vector<unsigned short int> number_student_skill;
  cin >> n;
  for (unsigned short int i = 1; i <= n; i++) {
    cin >> t;
    if (t == 1) {
      programming_students.push_back(i);
    }
    else if (t == 2) {
      maths_students.push_back(i);
    }
    else {
      pe_students.push_back(i);
    }
  }
  number_student_skill.push_back(programming_students.size());
  number_student_skill.push_back(maths_students.size());
  number_student_skill.push_back(pe_students.size());
  sort(number_student_skill.begin(), number_student_skill.end());
  w = number_student_skill.at(0);
  cout << w << endl;
  for (unsigned short int i = 0; i < w; i++) {
    cout << programming_students.at(i) << " " << maths_students.at(i) << " " << pe_students.at(i) << endl;
  }
  return 0;
}